introduction to seeder addon (#392)
harvester/docs: docs/advanced/seeder.md

---
sidebar_position: 7
sidebar_label: Seeder
title: "Seeder"
---

<head>
  <link rel="canonical" href="https://docs.harvesterhci.io/v1.1/advanced/seeder"/>
</head>

_Available as of v1.2.0_

The `harvester-seeder` addon, allows users to perform out-of-band operations on underlying nodes. 

Baremetal nodes that support redfish based access, the addon can also perform hardware discover, hardware events and correlate these with corresponding harvester nodes.

To get start the user needs to enable `harvester-seeder` addon from the Addons page

![](/img/v1.2/vm-import-controller/EnableAddon.png)

Once he addon is enabled, users can `Edit Config` for a specific host, to access `Out-Of-Band` definition page

![](/img/v1.2/seeder/EditConfig.png)

![](/img/v1.2/seeder/OutOfBandAccess.png)

`seeder` leverages `ipmi` for performing management operations on the underlying node hardware.

> `redfish` support is needed for optional hardware discovery and event detection capability.

Reviewer: Adding a hyperlink to `redfish` is better

## Performing power operations

Once an out-of-band config is defined for a node. Placing the node successfully into `Maintenance Mode`, allows users to Shutdown or Reboot the node.

![](/img/v1.2/seeder/ShutdownReboot.png)

A node once Shutdown, allows users to subsequently power it on.

![](/img/v1.2/seeder/PowerOn.png)

## Hardware event aggregation

If `Event` collection is enabled in `Out-of-Band Access`, then `seeder` leverages `redfish` to query the underlying hardware for information about component failures and fan temperatures.

> This information is correlated with Harvester nodes, and made available as kubernetes events

Reviewer: ```suggestion
This information is associated with Harvester nodes and can be used as Kubernetes events.
```

---
sidebar_position: 7
sidebar_label: Seeder
title: "Seeder"
---

<head>
  <link rel="canonical" href="https://docs.harvesterhci.io/dev/advanced/seeder"/>
</head>
_Available as of v1.2.0_

> The `harvester-seeder` addon allows you to perform out-of-band operations on underlying nodes. 

Reviewer 2: ```suggestion
The `harvester-seeder` addon lets you perform out-of-band operations on underlying nodes. 
```

> For bare-metal nodes that support redfish-based access, this addon can also discover hardware and hardware events, and then associate the hardware with the corresponding Harvester nodes.

Reviewer 2: ```suggestion
This addon can also discover hardware and hardware events for bare-metal nodes that support redfish-based access and then associate the hardware with the corresponding Harvester nodes.
```

> To get started, you need to enable the `harvester-seeder` addon from the **Addons** page.

Reviewer 2: ```suggestion
You must enable the `harvester-seeder` addon from the **Addons** page to get started.
```

![](/img/v1.2/vm-import-controller/EnableAddon.png)

> Once the addon is enabled, find the desired host and click **Edit Config**, and then click the `Out-Of-Band Access` tab.

Reviewer 2: ```suggestion
Once the addon is enabled, find the desired host and select **Edit Config** and go to the **Out-Of-Band Access** tab.
```

![](/img/v1.2/seeder/EditConfig.png)

![](/img/v1.2/seeder/OutOfBandAccess.png)

`seeder` leverages `ipmi` to manage the underlying node hardware.

Hardware discovery and event detection require `redfish` support.

## Power operations

Once you've defined the out-of-band config for a node, you can put the node into `Maintenance` mode, which allows you to shut down or reboot the node as needed.

![](/img/v1.2/seeder/ShutdownReboot.png)

> If a node is shut down, you can also click **Power On** to power it on again:

Reviewer 2: ```suggestion
If a node is shut down, you can also select **Power On** to power it on again:
```
